f2py-f90wrap v0.2.15 now leads to  "Fatal Error: Cannot open module file xyz.mod’ for reading"

I am not sure this is a `f90wrap` issue _per se_, but want to report it in case others bump into this issue. It may be a side effect of `numpy f2py` now relying on `meson` to build stuff.

## Issue

I have a python wrapper in [jmp75/air2stream at commit 02fc71](https://github.com/jmp75/air2stream/tree/02fc7140fe358228eb9d94514759bf00fd940fdf) that builds fine in a conda environment that includes: 

```text
f90wrap                   0.2.13 
numpy                     1.26.0 
python                    3.9.16
```

I am trying to run the same wrapping build script into a new environment with:

```text
python                    3.12.5
f90wrap                   0.2.15
numpy                     2.1.1
meson                     1.5.1  # now required by numpy-f2py it seems
```

This fails at the step calling `f2py-f90wrap`:

```text
[2/7] Compiling Fortran object _a2s.cpython-312-x86_64-linux-gnu.so.p/f90wrap_AIR2STREAM_MODULES.f90.o
FAILED: _a2s.cpython-312-x86_64-linux-gnu.so.p/f90wrap_AIR2STREAM_MODULES.f90.o 
gfortran -I_a2s.cpython-312-x86_64-linux-gnu.so.p -I. -I.. -I/home/abcdef/miniforge3/envs/a2s/lib/python3.12/site-packages/numpy/_core/include -I/home/abcdef/miniforge3/envs/a2s/lib/python3.12/site-packages/numpy/f2py/src -I/home/abcdef/miniforge3/envs/a2s/include/python3.12 -fvisibility=hidden -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -O3 -fPIC -J_a2s.cpython-312-x86_64-linux-gnu.so.p -o _a2s.cpython-312-x86_64-linux-gnu.so.p/f90wrap_AIR2STREAM_MODULES.f90.o -c ../f90wrap_AIR2STREAM_MODULES.f90
../f90wrap_AIR2STREAM_MODULES.f90:4:9:

    4 |     use commondata, only: commondata_n_par => n_par
      |         1
Fatal Error: Cannot open module file ‘commondata.mod’ for reading at (1): No such file or directory
compilation terminated.
[3/7] Compiling Fortran object _a2s.cpython-312-x86_64-linux-gnu.so.p/_a2s-f2pywrappers.f.o
[4/7] Compiling Fortran object _a2s.cpython-312-x86_64-linux-gnu.so.p/f90wrap_toplevel.f90.o
[5/7] Compiling C object _a2s.cpython-312-x86_64-linux-gnu.so.p/6a58a1f4e765b29ba60324f015623347fad0725c_.._.._f2py_src_fortranobject.c.o
[6/7] Compiling C object _a2s.cpython-312-x86_64-linux-gnu.so.p/_a2smodule.c.o
ninja: build stopped: subcommand failed.
```

--- a/README.md
+++ b/README.md
@@ -192,6 +192,44 @@ Notes
 - Arrays of derived types are currently not fully supported: a workaround is provided for 1D-fixed-length arrays, i.e. `type(a), dimension(b) :: c`. In this case, the super-type `Type_a_Xb_Array` will be created, and the array of types can be accessed through `c.items`. Note that dimension b can not be `:`, but can be a parameter.
 - Doxygen documentation in Fortran sources are parsed and given as docstring in corresponding python interfaces. Doxygen support is partial and keyword support is limited to `brief`, `details`, `file`, `author` and `copyright`.
 
+Troubleshooting
+---------------
+
+### NumPy 2.0+ and Meson Backend Issues
+
+Starting with NumPy 2.0, the `f2py` build system transitioned from `distutils`
+to `meson`. This can cause issues with f90wrap, particularly when Fortran module
+files (`.mod` files) cannot be found during compilation.
+
+**Symptom**: Errors like "Cannot open module file 'modulename.mod' for reading"
+during the f2py-f90wrap compilation step.
+
+**Workarounds**:
+
+1. **Set FFLAGS to include current directory** (recommended):
+   ```bash
+   export FFLAGS="-I$(pwd)"
+   f2py-f90wrap -c -m _mymodule f90wrap_*.f90 *.o
+   ```
+
+2. **Use the `--build-dir` option with f2py-f90wrap**:
+   ```bash
+   f2py-f90wrap --build-dir build -c -m _mymodule f90wrap_*.f90 mymodule.f90
+   ```
+   Note: When using `--build-dir`, pass the original `.f90` source files instead
+   of pre-compiled `.o` files, as the meson backend needs to compile them in the
+   build directory where the `.mod` files will be generated.
+
+3. **Downgrade to NumPy < 2.0** (temporary):
+   ```bash
+   pip install 'numpy<2.0'
+   ```
+
+**Note**: The `f2py-f90wrap` script includes patches to help with the meson
+backend, including automatic handling of include paths and library directories
+when using `--build-dir`. If you encounter issues, please report them at
+https://github.com/jameskermode/f90wrap/issues
+
 How f90wrap works
 -----------------
 

